ReSwift/CoreTypes/Store.swift

```diff
@@ -25,18 +25,23 @@ open class Store<State: StateType>: StoreType {
 
     /*private (set)*/ public var state: State! {
         didSet {
-            subscriptions = subscriptions.filter { $0.subscriber != nil }
+            var filtered = subscriptions
             subscriptions.forEach {
```

Comment:
I think this could be written as:
```swift
subscriptions = subscriptions.filter {
    if $0.subscriber != nil {
        $0.newValues(oldState: oldValue, newState: state)
        return true
    }
    return false
}
```
instead of assigning a variable, filtering, and then reassigning the variable

Reply:
I try to avoid side-effects in "pure" functions such as `filter`, `map`, etc. I will, however, test this out to see if its faster, as if theres some magic in `filter` that improves the performance its probably worth it.

Reply:
I too like to put side-effects outside, but from that point of view, I actually prefer the old `.filter` & `.forEach` sequence to the newly proposed approach which looks oddly imperative next to the rest of the code base. But the old code has a complexity of O(2n) in the worst case, while the new imperative code and @JoeCherry's side-effect-ful filter should be O(n), only, am I right?

My bet is on `.filter` with side effect being faster, because `Set.remove` takes additional time (the docs say it's an amortized complexity of O(1), so would this turn @mjarvis approach into O(n+1)? 🤔)

Reply:
Performance tests show no recordable difference between this PR and @JoeCherry's `filter` side-effect suggestion.

Reply:
I think I'm happy merging this, as its gotten a few eyes on it to make sure I haven't caused any regressions. Just need an approved review from someone :)

Reply:
Cool, it looks good the way it is